### PR TITLE
fix: cater for edge case base branches that dont exist:IN-1116

### PR DESCRIPTION
**Pull request by edison-vflow**
* IN-1116

### Brief description. What is this change?

* Cater for base branches that expire or when we rebase and the ancestor of current branch changes from what it was.

### Related PR 

https://github.com/voiceflow/creator-app/pull/7703

#### Tests

- [x]  See related PR

### CODE REVIEW

**Review by Xavientois — APPROVED** [review]

**Review by Xavientois — APPROVED** [review]

File: src/commands/utils/compute_base_revision.yml [hunk 1]
```diff
@@ -22,6 +22,11 @@ steps:
         # If we are on master, use the previous commit as the base
         if [[ $GIT_BRANCH == "trying" ]] || [[ $GIT_BRANCH == "staging" ]]; then
           BASE="master"
+        else
+          if [ -z "$(git rev-list $GIT_BRANCH  | grep $BASE_REVISION 2> /dev/null)" ]; then
```
Comment (xavidop) [inline src/commands/utils/compute_base_revision.yml:26]: can you add a comment explaining what does this line?

**Review by xavidop — CHANGES_REQUESTED** [review]

File: src/commands/utils/compute_base_revision.yml (hunk 1 above)
Comment (theProf) [reply src/commands/utils/compute_base_revision.yml:26]: this looks like a way of testing if a commit exists? you may want to try:

`git rev-parse -q --verify "$BASE_REVISION^{commit}"`

**Review by theProf — CHANGES_REQUESTED** [review]

File: src/commands/utils/compute_base_revision.yml (hunk 1 above)
Comment (edison-vflow) [reply src/commands/utils/compute_base_revision.yml:26]: Sure @xavidop

Comment (edison-vflow) [reply src/commands/utils/compute_base_revision.yml:26]: @theProf , I did try a lot of permutations of verifying that a commit exists.
I basically need something that will tell me that the base revision we want to use for that branch exists before we use it.
the -q option silences any output making it unusable.
The other thing is that we need the base itself so checking if the base exists in a branch is the simplest way

Comment (edison-vflow) [reply src/commands/utils/compute_base_revision.yml:29]: @xavidop , comments added

**Review by xavidop — APPROVED** [review]

File: src/commands/utils/compute_base_revision.yml (hunk 1 above)
Comment (theProf) [reply src/commands/utils/compute_base_revision.yml:29]: >  the -q option silences any output making it unusable.

that's not true by the manual and trial, but go with what's working for you to unblock

>         -q, --quiet
>            Only meaningful in --verify mode. Do not output an error message if the first argument is not a valid object name; instead exit
>            with non-zero status silently. SHA-1s for valid object names are printed to stdout on success.

```ogrady@Gregs-MacBook-Pro creator-app % git rev-parse -q --verify 785746976712c66a02b0cd1308a2e097b5a6586f^{commit}
785746976712c66a02b0cd1308a2e097b5a6586f
ogrady@Gregs-MacBook-Pro creator-app % git rev-parse -q --verify 785746976712c66a02b0cd1308a2e097b5a6586b^{commit}
ogrady@Gregs-MacBook-Pro creator-app %
```

**Review by theProf — APPROVED** [review]

**Pull request merged**